### PR TITLE
Feature/daniel/profile model

### PR DESCRIPTION
**Pull request by cimthog**

### CODE REVIEW

File: server/controller/profile.contoller.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
+import Profile from '../models/profile.model';
+
+// Create and Save a new Profile
+exports.create = (req, res) => {
+    // Validate request
+    if (!req.body.city || !req.body.country || !req.body.email
```
Comment (enyeInc) [inline server/controller/profile.contoller.js]: pull the logic in this if statement into a function that checks for the appropriate keys

File: server/controller/profile.contoller.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
+import Profile from '../models/profile.model';
+
+// Create and Save a new Profile
+exports.create = (req, res) => {
```
Comment (enyeInc) [inline server/controller/profile.contoller.js]: destrcuture `req.body`. Do that in the rest of this file. Also remember to destructure your `error` object as well
```
const { city, country ... } = req.body;
```

File: package.json [hunk 3]
```diff
@@ -32,21 +35,23 @@
     "eslint-plugin-promise": "^4.0.1",
     "eslint-plugin-react": "^7.11.1",
     "eslint-plugin-standard": "^4.0.0",
-    "jest": "^24.7.1",
+    "jest": "^24.8.0",
     "mocha": "^6.1.2",
     "sinon": "^7.3.1"
   },
   "dependencies": {
-    "@babel/core": "^7.4.3",
```
Comment (enyeInc) [inline package.json]: dont you need `@babel/core` to be able to run `@babel/register`?

File: server/index.js [hunk 4]
```diff
@@ -8,21 +8,31 @@ const nextApp = next({ dev });
...
+
 // Configure DB
 const db = require('../config/keys').mongoURI;
 
+// Connect to MongoDB
+mongoose
+    .connect(db, { useNewUrlParser: true })
+    .then()
+    .catch();
+
```
Comment (enyeInc) [inline server/index.js:22]: move the mongoose connect logic into its own file then import it here

File: server/controller/profile.contoller.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
+import Profile from '../models/profile.model';
```
Comment (enyeInc) [inline server/controller/profile.contoller.js]: go through this file and convert each function to leverage `async/await` pattern (https://javascript.info/async-await)

File: package.json [hunk 6]
```diff
@@ -32,21 +35,23 @@
...
+    "@shelf/jest-mongodb": "^1.0.1",
     "@zeit/next-css": "^1.0.1",
     "antd": "^3.16.3",
     "auth0-js": "^9.10.2",
     "babel-jest": "^24.7.1",
+    "@babel/register": "^7.4.4",
```
Comment (enyeInc) [inline package.json]: `@babel/register` is already in your dev dependencies. delete it from here

File: server/controller/profile.contoller.js (hunk 5 above)
Comment (uimarshall) [inline server/controller/profile.contoller.js]: We already have a folder for models folder in the repo.No need to create another one. Your profile.model should go to this folder.

File: server/controller/profile.contoller.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
```
Comment (uimarshall) [inline server/controller/profile.contoller.js]: The controller folder and routes is more or less the same, use routes/api/profile.js to keep it simple.
The routes folder should be at the same level with the models folder, remove it from the server folder.

Comment (cimthog) [reply server/controller/profile.contoller.js]: done, please review

File: package.json (hunk 6 above)
Comment (cimthog) [reply package.json]: removed

File: server/controller/profile.contoller.js (hunk 5 above)
Comment (cimthog) [reply server/controller/profile.contoller.js]: done!

**Review by justiceotuya — CHANGES_REQUESTED** [review]
your package.json and server.js has a merge conflict. pull from master, solve the merge conflict and push back to this branch

**Review by justiceotuya — APPROVED** [review]

File: models/profile.model.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    },
+    following: {
+        default: 0,
+        type: Number,
+    },
+    groups: {
+        default: 0,
+        type: Number,
+    },
+    image: {
```
Comment (uimarshall) [inline models/profile.model.js:36]: Am curious about why you use contentType and buffer for image

File: models/profile.model.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+const profileSchema = new Schema({
+    city: {
+        required: true,
+        type: String,
+    },
+    country: {
+        required: true,
+        type: String,
+    },
+    email: {
```
Comment (uimarshall) [inline models/profile.model.js:14]: Will lowercase:true for email makes whatever the user type in to be lowercase? or it fails if the user put space or Capitalize?

File: server/index.js [hunk 10]
```diff
@@ -11,19 +14,26 @@ const handle = nextApp.getRequestHandler();
...
+mongoose.connect(db, { useNewUrlParser: true });
+
+mongoose.connect(db, { useNewUrlParser: true });
 
 nextApp.prepare().then(() => {
     // express code here
     const app = express();
+    // bodyParser Middleware
     app.use(bodyParser.json());
-    app.use(bodyParser.urlencoded({ extended: true }));
```
Comment (uimarshall) [inline server/index.js:20]: Am curious about why you include this line: app.use(bodyParser.urlencoded({ extended: true }));

**Comment by enyeInc** [comment]
@cimthog close this PR, create a new off master and add your changes to that branch. There is no reason why you should have over 900+ changes in this PR.

File: models/profile.model.js (hunk 9 above)
Comment (cimthog) [reply models/profile.model.js:14]: it will save the email in lowercase

File: server/index.js (hunk 10 above)
Comment (cimthog) [reply server/index.js:20]: testing on postman, i noticed that bodyparser.json() isn't working

**Pull request closed without merging**